src/lib/services/feature-tag-service.ts

```diff
@@ -129,12 +130,20 @@ class FeatureTagService {
             await this.tagStore.getTag(tag.type, tag.value);
         } catch (error) {
             if (error instanceof NotFoundError) {
-                await this.tagStore.createTag(tag);
-                await this.eventStore.store({
-                    type: TAG_CREATED,
-                    createdBy: userName,
-                    data: tag,
-                });
+                try {
+                    await this.tagStore.createTag(tag);
+                    await this.eventStore.store({
+                        type: TAG_CREATED,
+                        createdBy: userName,
+                        data: tag,
+                    });
+                } catch (err) {
+                    if (err.code === FOREIGN_KEY_VIOLATION) {
+                        throw new BadDataError(
+                            `Tag type '${tag.type}' does not exist`,
+                        );
+                    }
+                }
```

Comment:
I'm not a big fan of nested try catches, the problem IMO is the getTag throwing an exception at the store level rather than returning `undefined`. Alternatively, we can move handling the `FOREIGN_KEY_VIOLATION` to the store level.

Reply:
From a consistency point of view, we're catching FK violations at the service layer: 

1. https://github.com/Unleash/unleash/blob/0b0191a2f746a146e7b731714aff7fc9415ec43a/src/lib/services/api-token-service.ts#L241
2. https://github.com/Unleash/unleash/blob/0b0191a2f746a146e7b731714aff7fc9415ec43a/src/lib/services/feature-toggle-service.ts#L456

Reply:
I feel like cleaning up the usage and handling it in the service layer is probably better. Does that mean you'd like to swap to undefined as the return value at the store level? That feels fair, what do we do in the feature store?

Reply:
At the feature-store we're also throwing the exception: https://github.com/Unleash/unleash/blob/openapi-tags/src/lib/db/feature-strategy-store.ts#L370-L372

So, now I'm unsure... Even if returning `undefined` and handling that at the service layer seems like a good thing to do, I also like to keep the code consistent and maybe change this as a whole later (if we decide to). 

I rather not have a try-catch inside a catch block, so I'm considering moving the FK handling to the store layer (although that also breaks consistency)

Reply:
+1 to try to keep consistency for now until we can apply a better generic solution, as long as we're aware that this is something we should try to improve soon-ish. Returning `undefined` in that scenario and handling in the service layer seems OK to me.